com.unity.ml-agents/Runtime/Policies/BrainParameters.cs

```diff
@@ -69,49 +53,16 @@ public ActionSpec ActionSpec
             }
         }
 
-        /// <summary>
-        /// (Deprecated) The number of possible actions.
-        /// </summary>
-        /// <remarks>The size specified is interpreted differently depending on whether
-        /// the agent uses the continuous or the discrete actions.</remarks>
-        /// <value>
-        /// For the continuous actions: the length of the float vector that represents
-        /// the action.
-        /// For the discrete actions: the number of branches.
-        /// </value>
-        [Obsolete("VectorActionSize has been deprecated, please use ActionSpec instead.")]
-        [FormerlySerializedAs("vectorActionSize")]
-        public int[] VectorActionSize = new[] { 1 };
```

Comment:
Unfortunately, I think we need to keep these around so that old assets can still be loaded.

Reply:
What do you mean by old assets ? What functionality are we trying to preserve ?

Reply:
To be clear, for v2, I was thinking refactoring BrainParameters and BehaviorParameters completely, so maybe it is best to set expectations early and tune down the scope I had in mind.

Reply:
If you have a scene or prefab from v1.6.0 or earlier, the `BrainParameters` have `VectorActionSize` and `VectorActionSpaceType` fields on them. When you first open this in 1.7.0 or later, we have to read the values from the old field and fill in the ActionSpec (m_ActionSpec) with them. The work happens in the UpdateToActionSpec() method, which is called in OnBeforeSerialize() and OnAfterDeserialize().

So if we remove these and a user loads a scene from <=1.6.0, their Agent will have no actions. In my opinion, this is a big pain for upgrading and we shouldn't do it. I'm willing to be convinced otherwise, but I think we need a bigger discussion on it.

That being said, I don't love that we have to keep the fields around (and public!) just so that they can be read once. It would be worth following up with `#devs-serialization` to see if there's a better way we can do this while hiding the fields. If nothing else, we can rename them to e.g. `DeprecatedDoNotUseVectorActionSize`.

Reply:
Also, these fields are used in a bunch of places that you didn't update, so this doesn't compile. I'll update yamato tests to run on PRs targeting the staging branch.